src/lib.rs

```diff
@@ -0,0 +1,166 @@
+mod fetcher;
```

Comment:
I personally don't really like to create implementations in `lib.rs`, but there are no written rules about this.  Just if the project grows it can become a bit cluttered.  I would consider creating `config.rs` and `error.rs` containing appropriate structs.

Also, I would say that it is more common for crates to export `Error` as opposed to `DatabaseError` as any importer can import `use xxx::Error as DatabaseError`.